### PR TITLE
refactor(editor): Migrate `App.vue` to composition api (no-changelog)

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
## Summary
Migrating our root component to composition API

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/App.vue [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+	logHiringBanner();
+	void useExternalHooks().run('app.mount');
+	loading.value = false;
+});
+
+defineExpose({
+	...globalLinkActions,
+	...historyHelper,
+	...toast,
+	...externalHooks,
```
Comment (tomi) [inline packages/editor-ui/src/App.vue:59]: Are these used somewhere via a reference to the app component?

Comment (MiloradFilipovic) [reply packages/editor-ui/src/App.vue:59]: No, not by reference but we need to initialize them. Will remove the `defineExpose`

**Review by RicardoE105 — COMMENTED** [review]
Just left a couple of comments that are NOT blockers! Aside from that nice work.

**Comment by MiloradFilipovic** [comment]
> Just left a couple of comments that are NOT blockers! Aside from that nice work.

Wrong PR @RicardoE105 :D

**Review by RicardoE105 — APPROVED** [review]
Nice to see we removed more files than we added 🔥

**Comment by janober** [comment]
Got released with [`n8n@1.57.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.57.0)

**Pull request merged**